src/core/components/InstallButton/index.js

```diff
@@ -37,15 +43,25 @@ export class InstallButtonBase extends React.Component {
   render() {
     const {
       addon,
-      clientSupportsAddons,
+      clientApp,
       className,
       hasAddonManager,
       i18n,
+      isCompatibleWithUserAgent,
       size,
+      userAgentInfo,
     } = this.props;
     const useButton = hasAddonManager !== undefined && !hasAddonManager;
     let button;
-    const buttonIsDisabled = !clientSupportsAddons();
+
+    // Test add-on capability (is the client Firefox?) and compatibility
+    // (is the client a valid version of Firefox to run this add-on?).
+    const { maxVersion, minVersion } = getCompatibleVersions({
```

Comment:
Totally true. Good call; I like it.

Like you said I'll keep the functions separate in the module, but I'll add `getClientCompatibility`. 👍